protocol/v2/qbft/storage/ibft_store.go

```diff
@@ -32,17 +36,63 @@ type ParticipantsRangeEntry struct {
 	Identifier convert.MessageID
 }
 
+type Quorum struct {
+	Signers     []spectypes.OperatorID
+	Committee   []spectypes.OperatorID
+	ValidatorPK spectypes.ValidatorPK // optional payload
+}
+
+func (q *Quorum) ToBitMask() OperatorsBitMask {
+	if len(q.Committee) > maxCommitteeSize || len(q.Signers) > maxCommitteeSize || len(q.Signers) > len(q.Committee) {
+		panic(fmt.Sprintf("invalid signers/quorum size: %d/%d", len(q.Committee), len(q.Signers)))
+	}
+
+	bitmask := OperatorsBitMask(0)
+	i, j := 0, 0
+	for i < len(q.Signers) && j < len(q.Committee) {
+		if q.Signers[i] == q.Committee[j] {
+			bitmask |= 1 << uint(j) // #nosec G115 -- j cannot exceed maxCommitteeSize
+			i++
+			j++
+		} else if q.Signers[i] < q.Committee[j] {
+			i++
+		} else { // A[i] > B[j]
+			j++
+		}
+	}
+
+	return bitmask
+}
+
+type OperatorsBitMask uint16
```

Comment:
Would be nice to describe the format in details (what bits correspond to what), in other words
- what this bit mask is
- how it's meant to be constructed

from what I've gathered, it's meant to "encode" only operators that have signed, if so - perhaps `SignersBitMask` would be a more descriptive name